compiler/src/abi/mod.rs

```diff
@@ -0,0 +1,23 @@
+use crate::errors::CompileError;
+use vyper_parser as parser;
+
+mod json_builder;
+
+/// Builds a JSON ABI from the source file.
+///
+/// This API should be rethought as it only provides the ABI for the first contract in the
+/// source file. See: https://github.com/ethereum/rust-vyper/issues/12
+/// TODO: Improve the JSON ABI builder API.
+pub fn build(src: &str) -> Result<String, CompileError> {
+    let tokens = parser::get_parse_tokens(src).unwrap();
+    let vyp_module = parser::parsers::file_input(&tokens[..]).unwrap().1.node;
+    let contracts = json_builder::module(&vyp_module)?;
+
+    if let Some(contract) = contracts.get(0) {
+        return Ok(serde_json::to_string(&contract.functions)?);
+    }
```

Comment:
No reason other than just wanting a simple API to create a minimal compiler binary with. I've since removed the binary component, and now think it would be better to just return a map `contract name -> JSON ABI`.

I think it might be good to add an additional function like this:

``` rust
pub fn build_ethpm(src: &str) -> Result<EthPMDesc, CompileError> { ... }
```

in a separate PR addressing the [issue](https://github.com/ethereum/rust-vyper/issues/12) mentioned above.